packages/permissionless/accounts/kernel/utils/getNonceKey.ts

```diff
@@ -0,0 +1,36 @@
+import { concatHex, maxUint16, pad, toHex } from "viem"
+import { VALIDATOR_MODE, VALIDATOR_TYPE } from "../constants"
+import {
+    KERNEL_VERSION_TO_ADDRESSES_MAP,
+    type KernelVersion
+} from "../signerToEcdsaKernelSmartAccount"
+
+export const getNonceKeyWithEncoding = (
+    accountVerion: KernelVersion,
+    nonceKey = 0n
+) => {
+    if (accountVerion === "0.2.2") {
+        return nonceKey
+    }
+
+    if (nonceKey > maxUint16)
+        throw new Error(
+            `nonce key must be equal or less than 2 bytes(maxUint16) for Kernel version ${accountVerion}`
+        )
+
+    const ecdsaValidatorAddress =
```

Comment:
User might have passed a different ESDSA_VALIDATOR address in the `signerToEcdsaKernelSmartAccount` we might want to pass that down here.

Reply:
ah yes. Sure will make this change